extensions/peerjs.js

```diff
@@ -0,0 +1,957 @@
+                        }
+                    },
+                    {
+                        opcode: "doesPeerHaveChannel",
+                        blockType: Scratch2.BlockType.BOOLEAN,
+                        text: "does [ID] have channel [CHANNEL]?",
+                        arguments: {
+                            ID: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "B"
+                            },
+                            CHANNEL: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "blueberry"
+                            },
+                        }
+                    },
+                    {
+                        opcode: "sendMessageToPeer",
+                        blockType: Scratch2.BlockType.COMMAND,
+                        text: "send [MESSAGE] to peer [ID] using channel [CHANNEL]",
+                        arguments: {
+                            MESSAGE: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "hello world"
+                            },
+                            ID: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "B"
+                            },
+                            CHANNEL: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "default"
+                            },
+                        }
+                    },
+                    {
+                        opcode: "openNewPeerChannel",
+                        blockType: Scratch2.BlockType.COMMAND,
+                        text: "open a new channel [CHANNEL] with peer [ID] and enforce order? [ORDERED]",
+                        arguments: {
+                            ID: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "B"
+                            },
+                            CHANNEL: {
+                                type: Scratch2.ArgumentType.STRING,
+                                defaultValue: "blueberry"
+                            },
+                            ORDERED: {
+                                type: Scratch2.ArgumentType.BOOLEAN,
+                                defaultValue: false
+                            },
+                        }
+                    },
+                ],
+                menus: {},
+            }
+        }
+
+        // Internal functions that aren't mapped to blocks
+        handleChannelOpen(conn, chan) {
+            conn.channels.set(chan.label, {chan, data: ""});
+        }
+
+        handleChannelClose(conn, chan) {
+            conn.channels.delete(chan.label);
+        }
+
+        handleChannelError(conn, chan, err) {
+            const self = this;
```

Comment:
Why use `self` if `this` still works fine? It doesn't really simplify things in the way you're using it.
Ex. `this.peer.errorInfo` still works.